truffle/com.oracle.truffle.api.instrumentation/src/com/oracle/truffle/api/instrumentation/InstrumentationHandler.java

```diff
@@ -536,17 +538,18 @@ protected void visitInstrumented(Node node, SourceSection section) {
     }
 
     /**
-     * Instrumenter implementation for {@link TruffleInstrument}.
+     * Instrumenter implementation for use by {@linkplain TruffleInstrument external clients} of
+     * instrumentation.
      */
-    final class InstrumentationInstrumenter extends AbstractInstrumenter {
+    final class ClientInstrumenter extends AbstractInstrumenter {
```

Comment:
I understand the internal consistency behind the name, but it unfortunately leads to a word that visually scans very poorly and is distracting.  Fortunately it is not public.  How about changing both names:  `LanguageClientInstrumenter` and `InstrumentClientInstrumenter`?

The source of the awkwardness is the name `Instrumenter`, which now that I've thought about it more should really be called something like `Binder`.  This would be much more descriptive, since it creates instances of `EventBinding`, but that's a breaking change.